### DISCUSSION TITLE
Multiple views like facets

### DISCUSSION THREAD

**Discussion by ArnaudRo** [visgl/deck.gl#6525] [Q&A] [answered]
I'd like to design a webapp with facet maps sharing same viewState using deckgl multiview feature. Same dataset would be presented in different layers/mapview thanks to LayerFilter etc
Trying to adapt the minimap example

```
import React, {useState, useCallback} from 'react';
import {StaticMap} from 'react-map-gl';
import {MapView} from '@deck.gl/core';
import DeckGL from '@deck.gl/react';
import {BASEMAP} from '@deck.gl/carto';

const views = [
  new MapView({id: 'facetMap1_1', x:0,y:0, width:'50%', height:'50%',...}),
  new MapView({id: 'facetMap1_2', x:'50%',y:0, width:'50%', height:'50%',...}),
  new MapView({id: 'facetMap2_1', x:0,y:'50%', width:'50%', height:'50%',...}),
  new MapView({id: 'facetMap2_2', x:'50%',y:'50%', width:'50%', height:'50%',...})
];

function App() {
  return (
    <DeckGL views={views} initialViewState={...} layers={...} >
       <MapView id='facetMap1_1'><StaticMap  mapStyle={BASEMAP.POSITRON}/></MapView>
       <MapView id='facetMap1_2'><StaticMap  mapStyle={BASEMAP.POSITRON}/></MapView>
       <MapView id='facetMap2_1'><StaticMap  mapStyle={BASEMAP.POSITRON}/></MapView>
       <MapView id='facetMap2_2'><StaticMap  mapStyle={BASEMAP.POSITRON}/></MapView>
    </DeckGL>
  );
}
```
As i could have various number of facets (up to 4x4 but 2x3 is also possible) wanted to simplify code creating views array and JSX part (<MapView etc...). 
I do not know how to access "MapViews id" as children of DeckGL component
Could someone help me ?
Thanksfully

**Comment by Pessimistress** [comment, marked as answer]
Not sure what you are asking, could this work?

```jsx
return (
  <DeckGL views={views} initialViewState={...} layers={...} >
    { views.map(({id}) => <MapView key={id} id={id}><StaticMap  mapStyle={BASEMAP.POSITRON}/></MapView> }
  </DeckGL>
);

**Comment by ArnaudRo** [comment]
Thank you. it works as i'd like to. Simply i had a confusion between views (propName) and view (propValue) and tryied (id)=> not ({id}=> in the function

**Comment by ArnaudRo** [comment]
ok, so "maplibre-gl.js:35 WARNING: Too many active WebGL contexts. Oldest context will be lost.
r._setupPainter @ maplibre-gl.js:35" and "Error: Framebuffer error 36061" ...

**Reply by Pessimistress** [reply]
There is a limit to how many WebGL contexts you can have. The limit is set by your browser and hardware. Mapbox/Maplibre creates one context per map and that's not something deck.gl controls.

**Comment by ArnaudRo** [comment]
12 views is OK zoom in-zoom out, 16 views in one DeckGL is a fail
